test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -19,18 +18,21 @@ function ($, Q, Expect, $sandbox, DeviceFingerprint) {
       window.postMessage(JSON.stringify(message), '*');
     }
 
-    function mockUserAgent(userAgent) {
-      navigator = new Object(); // eslint-disable-line no-global-assign
-      navigator.__proto__ || (navigator.__proto__ = _navigator);
-      navigator.__defineGetter__('userAgent', function () { return userAgent; });
+    function mockUserAgentCheck(userAgent) {
+      spyOn(DeviceFingerprint, '__getUserAgent').and.callFake(function() {
```

Comment:
The prev impl of this was wrong. The reset in afterEach dint happen correctly. Seems like it is not good to modify window.navigator for test and hence moved getting userAgent to a separate function to mock it correctly.

Reply:
Glad you find this issue!